ts/javascript/initializers.test.ts

```diff
@@ -117,9 +117,111 @@ Deno.test(
           })
             .addCheck("test", "npm run test")
             .addExecutable("start", "npm run start")
-            .addExecutable("build", "npm run build")
+            .addExecutable("build", "npm run build");
+        `,
+      );
+    }
+  },
+);
+
+Deno.test("adds the vite plugin if vite is in the devDependencies", () => {
+  const tempDir = Deno.makeTempDirSync();
+  Deno.writeTextFileSync(
+    join(tempDir, "package.json"),
+    JSON.stringify({
+      name: "somepackage",
+      description: "just some package",
+      devDependencies: {
+        vite: "*",
+      },
+    }),
+  );
+  const result = npmInitializer(tempDir);
+  assertEquals(result.tag, "ShouldRun");
+  if (result.tag === "ShouldRun") {
+    assertEquals(
+      result.makeTarget(),
+      outdent`
+        export const somepackage = garn.javascript.mkNpmProject({
+          description: "just some package",
+          src: ".",
+          nodeVersion: "18",
+        })
+          .add(garn.javascript.vite);
+      `,
+    );
+  }
+});
+
+Deno.test(
+  "doesn't add known vite scripts if superseeded by vite plugin",
+  () => {
+    const tempDir = Deno.makeTempDirSync();
+    Deno.writeTextFileSync(
+      join(tempDir, "package.json"),
+      JSON.stringify({
+        name: "somepackage",
+        description: "just some package",
+        scripts: {
+          build: "vite build",
+          dev: "vite",
+          foo: "test script",
+        },
+        devDependencies: {
+          vite: "*",
+        },
+      }),
+    );
+    const result = npmInitializer(tempDir);
+    assertEquals(result.tag, "ShouldRun");
+    if (result.tag === "ShouldRun") {
+      assertEquals(
+        result.makeTarget(),
+        outdent`
+          export const somepackage = garn.javascript.mkNpmProject({
+            description: "just some package",
+            src: ".",
+            nodeVersion: "18",
+          })
+            .add(garn.javascript.vite)
+            .addExecutable("dev", "npm run dev")
```

Comment:
Did we not want to provide the `dev` script?

Reply:
No, we don't. This was back from my change discussed in https://github.com/garnix-io/garn/pull/397#discussion_r1392144122, but now it's omitted again.